source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -644,6 +649,11 @@ void Filter::onUpstreamEvent(Network::ConnectionEvent event) {
         });
       }
     }
+    if (config_->maxConnectionDuration()) {
```

Comment:
I think the logic here is sound, but what do you think of moving it to onNewConnection (maybe moving it to the cc file).  If Envoy is say doing transparent proxying for a high rtt long haul connection, it could take a while to establish an upstream connection and get that upstream event so better start the timer as early as possible yeah?

Reply:
Moved initializeUpstreamConnection which is return by onNewConnection.

Reply:
The timing is right, but I think it'd be better to do in onNewConnection since it doesn't have to do with the upstream connection.

Reply:
Moved in onNewConnection.